src/Notepads/Services/DesktopExtensionService.cs

```diff
@@ -0,0 +1,135 @@
+    }
+
+    public static class DesktopExtensionService
+    {
+        public static AppServiceConnection InteropServiceConnection = null;
+        private static readonly ResourceLoader _resourceLoader = ResourceLoader.GetForCurrentView();
+
+        private static readonly string _commandLabel = "Command";
+        private static readonly string _failedLabel = "Failed";
+        private static readonly string _adminCreatedLabel = "AdminCreated";
+
+        public static async Task Initialize()
+        {
+            InteropServiceConnection = new AppServiceConnection()
+            {
+                AppServiceName = "DesktopExtensionServiceConnection",
+                PackageFamilyName = Package.Current.Id.FamilyName
+            };
+
+            InteropServiceConnection.RequestReceived += InteropServiceConnection_RequestReceived;
+            InteropServiceConnection.ServiceClosed += InteropServiceConnection_ServiceClosed;
+
+            AppServiceConnectionStatus status = await InteropServiceConnection.OpenAsync();
+            if (status != AppServiceConnectionStatus.Success && !await ApplicationView.GetForCurrentView().TryConsolidateAsync()) 
+                Application.Current.Exit();
+        }
+
+        private static async void InteropServiceConnection_RequestReceived(AppServiceConnection sender, AppServiceRequestReceivedEventArgs args)
+        {
+            var message = args.Request.Message;
+            if (!message.ContainsKey(_commandLabel) || !Enum.TryParse(typeof(CommandArgs), (string)message[_commandLabel], out var commandObj) ||
+                (CommandArgs)commandObj != CommandArgs.CreateElevetedExtension) return;
+
+            await CoreApplication.MainView.CoreWindow.Dispatcher.CallOnUIThreadAsync(() =>
+            {
+                if (message.ContainsKey(_adminCreatedLabel) && (bool)message[_adminCreatedLabel])
+                {
+                    NotificationCenter.Instance.PostNotification(_resourceLoader.GetString("TextEditor_NotificationMsg_AdminExtensionCreated"), 1500);
+                }
+                else
+                {
+                    NotificationCenter.Instance.PostNotification(_resourceLoader.GetString("TextEditor_NotificationMsg_AdminExtensionCreationFailed"), 1500);
+                }
+            });
+        }
+
+        private static void InteropServiceConnection_ServiceClosed(AppServiceConnection sender, AppServiceClosedEventArgs args)
+        {
+            return;
+        }
+
+        public static async Task SaveFileAsAdmin(string filePath, byte[] data)
+        {
+            if (InteropServiceConnection == null) await Initialize();
+
+            using (var adminConnectionPipeStream = new NamedPipeServerStream($"Local\\{Package.Current.Id.FamilyName}\\AdminWritePipe",
+                PipeDirection.InOut, 254, PipeTransmissionMode.Message, PipeOptions.Asynchronous))
+            {
+                if (!adminConnectionPipeStream.WaitForConnectionAsync().Wait(250))
```

Comment:
@JasonStein what could be the minimum ms we can wait here??

Reply:
Is it going to affect on startup?

Reply:
No, this would only apply when user is trying to modify a system file. Notepads tries to connect to the desktop extension component within 250ms. If it succeeds it sends the data to the component otherwise it prompts user to run the extension as adminstrator.